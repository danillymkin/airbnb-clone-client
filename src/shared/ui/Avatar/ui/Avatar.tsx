import React, { FunctionComponent } from 'react';
import Image from 'next/image';

export const Avatar: FunctionComponent = (): JSX.Element => {
  return (
    <Image
      className={'rounded-full'}
      src={'/images/avatar-fallback.jpg'}
      width={30}
      height={30}
      alt={'Avatar'}
    />
  );
};
