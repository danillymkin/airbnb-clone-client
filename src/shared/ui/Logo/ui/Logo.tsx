'use client';

import React, { FunctionComponent } from 'react';
import Image from 'next/image';

export const Logo: FunctionComponent = (): JSX.Element => {
  return (
    <Image
      alt={'Logo'}
      className={'block cursor-pointer'}
      height={31}
      width={100}
      src={'/images/logo.png'}
      priority
    />
  );
};
