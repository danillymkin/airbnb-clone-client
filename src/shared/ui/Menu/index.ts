export { Menu } from './ui/Menu/Menu';
export { MenuItem } from './ui/MenuItem/MenuItem';
