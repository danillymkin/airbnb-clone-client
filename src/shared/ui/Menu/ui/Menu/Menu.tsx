'use client';

import React, { FunctionComponent } from 'react';
import { Menu as HeadlessMenu } from '@headlessui/react';

interface MenuProps {
  trigger: React.ReactNode;
  children: React.ReactNode;
}

type Props = MenuProps;

export const Menu: FunctionComponent<Props> = ({
  trigger,
  children,
}): JSX.Element => {
  return (
    <HeadlessMenu as={'div'} data-testid={'menu'} className={'relative'}>
      <HeadlessMenu.Button>{trigger}</HeadlessMenu.Button>

      <HeadlessMenu.Items
        className={
          'absolute right-0 top-12 w-[40vw] overflow-hidden rounded-xl bg-white text-sm shadow-md md:w-52'
        }
      >
        {children}
      </HeadlessMenu.Items>
    </HeadlessMenu>
  );
};
