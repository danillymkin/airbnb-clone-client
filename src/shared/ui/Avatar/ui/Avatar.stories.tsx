import { Meta, StoryObj } from '@storybook/react';
import { Avatar } from '@/shared/ui/Avatar';

const meta: Meta = {
  title: 'UI/Avatar',
  component: Avatar,
};

export default meta;

type Story = StoryObj<typeof Avatar>;

export const Default: Story = {
  render: () => <Avatar />,
};
