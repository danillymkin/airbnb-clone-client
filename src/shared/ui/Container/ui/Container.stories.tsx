import { Meta, StoryObj } from '@storybook/react';
import { Container } from '@/shared/ui/Container';

const meta: Meta = {
  title: 'UI/Container',
  component: Container,
  parameters: {
    layout: 'fullscreen',
  },
};

export default meta;

type Story = StoryObj<typeof Container>;

export const Default: Story = {
  render: () => (
    <Container>
      <div>Children</div>
    </Container>
  ),
};
