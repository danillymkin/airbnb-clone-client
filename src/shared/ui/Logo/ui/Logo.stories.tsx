import { Meta, StoryObj } from '@storybook/react';
import { Logo } from '@/shared/ui/Logo';

const meta: Meta<typeof Logo> = {
  title: 'UI/Logo',
  component: Logo,
};

export default meta;

type Story = StoryObj<typeof Logo>;

export const Default: Story = {
  render: () => <Logo />,
};
