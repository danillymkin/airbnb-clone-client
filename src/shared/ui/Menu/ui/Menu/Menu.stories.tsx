import { Meta, StoryObj } from '@storybook/react';
import { Menu, MenuItem } from '@/shared/ui/Menu';

const meta: Meta = {
  title: 'UI/Menu',
  component: Menu,
  parameters: {
    layout: 'centered',
  },
};

export default meta;

type Story = StoryObj<typeof Menu>;

export const Default: Story = {
  render: () => (
    <Menu trigger={'Menu'}>
      <MenuItem>Item 1</MenuItem>

      <MenuItem>Item 2</MenuItem>
    </Menu>
  ),
};
