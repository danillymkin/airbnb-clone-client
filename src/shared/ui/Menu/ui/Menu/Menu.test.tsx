import { render, screen } from '@testing-library/react';
import { Menu, MenuItem } from '@/shared/ui/Menu';

describe('Menu', () => {
  it('should render menu', () => {
    render(
      <Menu trigger={''}>
        <MenuItem>item</MenuItem>
      </Menu>
    );
    const menu = screen.getByTestId('menu');
    expect(menu).toBeInTheDocument();
  });

  it('should render menu trigger', () => {
    const trigger = 'trigger';
    render(
      <Menu trigger={trigger}>
        <MenuItem>item</MenuItem>
      </Menu>
    );
    const menu = screen.getByRole('button');
    expect(menu).toHaveTextContent(trigger);
  });
});
