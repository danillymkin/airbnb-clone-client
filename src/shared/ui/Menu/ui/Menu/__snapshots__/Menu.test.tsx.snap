// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match snapshot 1`] = `
<div
  class="relative"
  data-headlessui-state=""
  data-testid="menu"
>
  <button
    aria-expanded="false"
    aria-haspopup="menu"
    data-headlessui-state=""
    id="headlessui-menu-button-:r2:"
    type="button"
  >
    trigger
  </button>
</div>
`;
